src/quo/components/navigation/top_nav/view.cljs

```diff
@@ -87,20 +91,35 @@
              :size                :small}
             avatar-props)]]])
 
+(defn- network-status-label
+  [now wallet-latest-update]
+  (when (t/after? now
+                  (t/plus wallet-latest-update (t/minutes 5)))
+    (let [units    [{:name :t/datetime-second-short :limit 60 :in-second 1}
+                    {:name :t/datetime-minute-mid :limit 3600 :in-second 60}
+                    {:name :t/datetime-hour-short-lowercase :limit 86400 :in-second 3600}
+                    {:name :t/datetime-day :limit nil :in-second 86400}]
+          time-ago (datetime/time-ago-long wallet-latest-update units)]
```

Comment:
Found this to be a bit hard to understand. I think it would be beneficial to separate time calculating for `time-ago` into separate function and provide a docstring explaining what is the output

Reply:
I agree 👍  we could decouple this logic and place it in `status-im.common.home.top-nav.view` ns. In that way, the component stays pure.